statsmodels/stats/contingency_tables.py

```diff
@@ -0,0 +1,1215 @@
+
+        fmt = float_format
+
+        headers = ["Statistic", "P-value", "DF"]
+        stubs = ["Symmetry", "Homogeneity"]
+        stat1, pvalue1, df1 = self.symmetry()
+        stat2, pvalue2, df2 = self.homogeneity()
+        data = [[fmt % stat1, fmt % pvalue1, '%d' % df1],
+                [fmt % stat2, fmt % pvalue2, '%d' % df2]]
+        tab = iolib.SimpleTable(data, headers, stubs, data_aligns="r",
+                                 table_dec_above='')
+
+        return tab
+
+
+class Table2x2(SquareTable):
+    """
+    Analyses that can be performed on a 2x2 contingency table.
+
+    Note that for the risk ratio, the analysis is not symmetric with
+    respect to the rows and columns of the contingency table.  The two
+    rows define population subgroups, column 0 is the number of
+    'events', and column 1 is the number of 'non-events'.
+
+    Parameters
+    ----------
+    table : array-like
+        A 2x2 contingency table
+    shift_zeros : boolean
+        If true, 0.5 is added to all cells of the table if any cell is
+        equal to zero.
+    """
+
+    def __init__(self, table, shift_zeros=True):
+
+        if (table.ndim != 2) or (table.shape[0] != 2) or (table.shape[1] != 2):
+            raise ValueError("Table2x2 takes a 2x2 table as input.")
+
+        super(Table2x2, self).__init__(table, shift_zeros)
+
+
+    @classmethod
+    def from_data(cls, data, shift_zeros=True):
+        """
+        Construct a Table object from data.
+
+        Parameters
+        ----------
+        data : array-like
+            The raw data, the first column defines the rows and the
+            second column defines the columns.
+        shift_zeros : boolean
+            If True, and if there are any zeros in the contingency
+            table, add 0.5 to all four cells of the table.
+        """
+
+        if isinstance(data, pd.DataFrame):
+            table = pd.crosstab(data.iloc[:, 0], data.iloc[:, 1])
+        else:
+            table = pd.crosstab(data[:, 0], data[:, 1])
+        return cls(table, shift_zeros)
+
+
+    @cache_readonly
+    def log_oddsratio(self):
+        """
+        The log odds ratio of the table.
+        """
+        f = self.table.flatten()
+        return np.dot(np.log(f), np.r_[1, -1, -1, 1])
+
+
+    @cache_readonly
+    def oddsratio(self):
+        """
+        The odds ratio of the table.
+        """
+        return self.table[0, 0] * self.table[1, 1] / (self.table[0, 1] * self.table[1, 0])
+
+
+    @cache_readonly
+    def log_oddsratio_se(self):
+        """
+        The asymptotic standard error of the estimated log odds ratio.
+        """
+        return np.sqrt(np.sum(1 / self.table))
+
+
+    @cache_readonly
+    def oddsratio_pvalue(self):
+        """
+        P-value for the null hypothesis that the odds ratio equals 1.
+        """
+        zscore = self.log_oddsratio / self.log_oddsratio_se
+        pvalue = 2 * stats.norm.cdf(-np.abs(zscore))
+        return pvalue
```

Comment:
based on #2605 and the proposed enhancements, I think we will need a `method` argument here so it would need to be a method and not a property.
Assuming we want to add different tests with choices similar to `proportion_confint`.

A `method` argument could be added at some later time, but switching from cached property to method would break backwards compatibility.

(Also: the more general version would allow the null hypothesis that odds ratio is equal to some value different from 1, for inferiority, superiority and similar tests.)